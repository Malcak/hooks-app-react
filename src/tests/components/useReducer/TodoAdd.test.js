import React from 'react';
import { shallow } from 'enzyme';
import { TodoAdd } from '../../../components/useReducer/TodoAdd';

describe('tests on the <TodoAdd /> component', () => {
  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);

  test('should display the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should not call the function "handleAddAll"', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} });
    expect(handleAddTodo).not.toHaveBeenCalled();
  });
  test('should call the function "handleAddAll".', () => {
    const value = 'Learn Nodejs';
    wrapper.find('input').simulate('change', {
      target: {
        name: 'description',
        value,
      },
    });
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalled();
    expect(handleAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false,
    });
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
