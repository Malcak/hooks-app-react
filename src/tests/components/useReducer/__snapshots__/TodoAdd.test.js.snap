// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on the <TodoAdd /> component should display the component correctly 1`] = `
<Fragment>
  <h4>
    Add TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="do..."
      type="text"
      value=""
    />
    <div
      className="d-grid gap-1"
    >
      <button
        className="btn btn-outline-success mt-1"
        type="submit"
      >
        Add
      </button>
    </div>
  </form>
</Fragment>
`;
